lib/index.js

```diff
@@ -4683,6 +4683,43 @@ addToplevelMethod('wrapCallback', function (f, /*optional*/mappingHint) {
 });
 
 /**
+ * Wraps a function that returns a promise, transforming it to a function
+ * which accepts the same arguments and returns a Highland Stream instead.
+ * The wrapped function keeps its context, so you can safely use it as a
+ * method without binding.
+ *
+ * @id wrapAsync
+ * @section Utils
+ * @name _.wrapAsync(f)
+ * @param {Function} f - the function that returns a promise
+ * @api public
+ *
+ * var Promise = require('bluebird');
+ *
+ * var resolve = _.wrapAsync(Promise.resolve);
+ * var reject = _.wrapAsync(Promise.reject);
+ *
+ * resolve([1, 2, 3]).apply(function (a, b, c) {
+ *  // a === 1
+ *  // b === 2
+ *  // c === 3
+ * });
+ *
+ * reject('boom').errors(function (err) {
+ *   // err === 'boom'
+ * });
+ */
+
+addToplevelMethod('wrapAsync', function (f) {
+    var stream = this;
+    return function () {
+        var self = this;
+        var args = slice.call(arguments);
```

Comment:
Cool! I've always assumed that `arguments` must be converted to a proper array before you can use it like this.